hack/tools/check-go-version.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+
+BASE_REF=${1:-main}
+GO_VER=$(sed -En 's/^go (.*)$/\1/p' "go.mod")
+OLDIFS="${IFS}"
+IFS='.' MAX_VER=(${GO_VER})
+IFS="${OLDIFS}"
+
+if [ ${#MAX_VER[*]} -ne 3 -a ${#MAX_VER[*]} -ne 2 ]; then
+    echo "Invalid go version: ${GO_VER}"
+    exit 1
+fi
+
+GO_MAJOR=${MAX_VER[0]}
+GO_MINOR=${MAX_VER[1]}
+GO_PATCH=${MAX_VER[2]}
+
+RETCODE=0
+
+check_version () {
+    whole=$1
+    file=$2
+    OLDIFS="${IFS}"
+    IFS='.' ver=(${whole})
+    IFS="${OLDIFS}"
+
+    if [ ${ver[0]} -gt ${GO_MAJOR} ]; then
+        echo "${file}: ${whole}: Bad golang version (expected ${GO_VER} or less)"
+        return 1
+    fi
+    if [ ${ver[1]} -gt ${GO_MINOR} ]; then
+        echo "${file}: ${whole}: Bad golang version (expected ${GO_VER} or less)"
+        return 1
+    fi
+    
+    if [ ${#ver[*]} -eq 2 ] ; then
+        echo "${file}: ${whole}: Version is good"
+        return 0
+    fi
+    if [ ${#ver[*]} -ne 3 ] ; then
+        echo "${file}: ${whole}: Badly formatted golang version"
+        return 1
+    fi
+
+    if [ ${ver[1]} -eq ${GO_MINOR} -a ${ver[2]} -gt ${GO_PATCH} ]; then
+        echo "${file}: ${whole}: Bad golang version (expected ${GO_VER} or less)"
+        return 1
+    fi
+    echo "${file}: ${whole}: Version is good"
+    return 0
+}
+
+echo "Looking at golang version: ${GO_VER}"
+
+for f in $(find . -name "*.mod"); do
+    v=$(sed -En 's/^go (.*)$/\1/p' ${f})
+    if [ -z ${v} ]; then
+        echo "${f}: Skipping, no version found"
+        continue
+    fi
+    if ! check_version ${v} ${f}; then
+        RETCODE=1
+    fi
+    old=$(git grep -ohP '^go .*$' "${BASE_REF}" -- "${f}")
```

Comment:
This regex stipulates ONE space after "go"... (beware of fragile regex's)
This is a nit, as this should always hold :-)

Reply:
Actually it stipulates one space after "go", then ANYTHING, including spaces. Since we're not "capturing" the version separately, it doesn't matter.

Reply:
It's managed by `go mod`, so I would hope it always holds.